pkg/lockfile/parse-maven-lock.go

```diff
@@ -67,20 +66,25 @@ func NewMavenRegistryAPIClient(url string) (DepFile, error) {
 	}, nil
 }
 
-func (m *MavenRegistryProject) Open(path string) (NestedDepFile, error) {
+func (m *MavenRegistryProject) Open(_ string) (NestedDepFile, error) {
 	panic("Should not be called")
 }
 
 func (m *MavenRegistryProject) Path() string {
 	return m.path
 }
 
+type MavenStringHolder struct {
```

Comment:
Why making it only specific to maven? We could have the same issue with other package managers down the line. Better make it generic like `StringHolder`, with proper documentations, test, constructors, etc.